Core/clim-basic/geometry/region-predicates.lisp

```diff
@@ -83,21 +134,80 @@
                (wn 0))
            (map-over-polygon-segments
             (lambda (x1 y1 x2 y2)
+              ;; Algorithm is not predictible for polygon edges - we
+              ;; need to test for them explicitly. -- jd 2019-09-27
+              (when (segment-contains-point-p x1 y1 x2 y2 x y)
+                (return-from region-contains-position-p t))
               (if (<= y1 y)
                   (when (and (> y2 y)
                              (> (is-left x1 y1 x2 y2 x y) 0))
                     (incf wn))
                   (when (and (<= y2 y)
                              (< (is-left x1 y1 x2 y2 x y) 0))
                     (decf wn))))
-            self)
+            region)
            (not (zerop wn))))))
 
+(defmethod region-contains-position-p ((self rectangle) x y)
+  (multiple-value-bind (x1 y1 x2 y2)
+      (rectangle-edges* self)
+    (and (coordinate-between* x1 x x2)
+         (coordinate-between* y1 y y2))))
+
+(defmethod region-contains-position-p ((self ellipse) x y)
+  (nest
+   (multiple-value-bind (cx cy) (ellipse-center-point* self))
+   (let ((alpha (or (ellipse-start-angle self) 0))
+         (omega (or (ellipse-end-angle self) (* 2 pi)))
+         (polar->screen (polar->screen self)))
+     (and (angle-contains-point-p alpha omega (- x cx) (- y cy))
+          (nest (multiple-value-bind (polar-x polar-y)
+                    (untransform-position polar->screen x y))
+                ;; FIXME we don't need to factor the additonal epsilon
+                ;; but rotated elliptoids are naively rendered in clx.
+                (multiple-value-bind (polar-dx polar-dy)
+                    (untransform-distance polar->screen 1 1))
+                (coordinate<= (+ (square polar-x) (square polar-y))
+                              (+ 1 (square polar-dx) (square polar-dy))))))))
+
+
+;;   REGION-INTERSECTS-REGION-P region1 region2
+;;
+;;        Returns nil if region-intersection of the two regions region1 and
+;;        region2 would be +nowhere+; otherwise, it returns t.
+;;
+;;        aka region1 und region2 sind nicht disjunkt  aka AB /= 0
```

Comment:
Translation: "that means region1 and region2 are not disjoint, that means A ∩ B ≠ ∅"

Reply:
fixed (I've left `aka` since it doesn't need translation)